include/LocklessRingBuffer.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * LocklessRingBuffer.h - LMMS wrapper for a lockless ringbuffer library
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LOCKLESSRINGBUFFER_H
+#define LOCKLESSRINGBUFFER_H
+
+#include "lmms_export.h"
+#include "../src/3rdparty/ringbuffer/include/ringbuffer/ringbuffer.h"
+#include "lmms_basics.h"
+#include <QMutex>
+#include <QWaitCondition>
+
+
+//! A convenience layer for a realtime-safe and thread-safe multi-reader ring buffer library.
+template <class T>
+class LocklessRingBufferBase
+{
+	template<class _T>
+	friend class LocklessRingBufferReader;
+public:
+	LocklessRingBufferBase(std::size_t sz) : m_buffer(sz)
+	{
+		m_buffer.touch();	// reserve storage space before realtime operation starts
+	}
+	~LocklessRingBufferBase() {};
+
+	std::size_t capacity() const {return m_buffer.maximum_eventual_write_space();}
+	std::size_t free() const {return m_buffer.write_space();}
+	void wakeAll() {m_notifier.wakeAll();}
+
+protected:
+	ringbuffer_t<T> m_buffer;
+	QWaitCondition m_notifier;
+};
+
+
+// The sampleFrame_copier is required because sampleFrame is just a two-element
+// array and therefore does not have a copy constructor needed by std::copy.
+class sampleFrame_copier
+{
+	const sampleFrame* src;
+public:
+	sampleFrame_copier(const sampleFrame* src) : src(src) {}
+	void operator()(std::size_t src_offset, std::size_t count, sampleFrame* dest)
```

Comment:
`srcOffset`

Reply:
Was there really a camel case thing in the conventions before? I often used underscore in local variable names before in all the other classes.
I feel like the conventions are a moving target... There could at least be an announcement when they change.

Reply:
They changed a lot recently. It earlier said: "Variable and method names begin with a lower case letter". @Veratil changed it to "SHOULD be camelCase". So it's indeed optional. ~~But clang-format will eradicate any style freedom anyways~~